### ISSUE TITLE
ICS Feed for calender returns exception

### ISSUE THREAD

**Issue by gglnx** [solspace/craft-calendar#159]
**Describe the bug or issue you're experiencing**

On opening `.icsUrl` I get fellowing exception for my calender:

```
yii\base\ErrorException: Undefined array key 1 in vendor/solspace/craft-calendar/packages/plugin/src/Library/Export/ExportCalendarToIcs.php:115
Stack trace:
#0 vendor/craftcms/cms/src/web/ErrorHandler.php(77): yii\base\ErrorHandler->handleError(2, 'Undefined array...', '/Users/dennis/P...', 115)
#1 vendor/solspace/craft-calendar/packages/plugin/src/Library/Export/ExportCalendarToIcs.php(115): craft\web\ErrorHandler->handleError(2, 'Undefined array...', '/Users/dennis/P...', 115)
#2 vendor/solspace/craft-calendar/packages/plugin/src/Library/Export/ExportCalendarToIcs.php(32): Solspace\Calendar\Library\Export\ExportCalendarToIcs->combineExportString(Object(Solspace\Calendar\Elements\Event), Object(Carbon\Carbon))
#3 vendor/solspace/craft-calendar/packages/plugin/src/Library/Export/AbstractExportCalendar.php(55): Solspace\Calendar\Library\Export\ExportCalendarToIcs->prepareStringForExport()
#4 vendor/solspace/craft-calendar/packages/plugin/src/Controllers/ApiController.php(40): Solspace\Calendar\Library\Export\AbstractExportCalendar->output()
#5 [internal function]: Solspace\Calendar\Controllers\ApiController->actionIcs()
#6 vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#7 vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#8 vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('ics', Array)
#9 vendor/craftcms/cms/src/web/Application.php(301): yii\base\Module->runAction('calendar/api/ic...', Array)
#10 vendor/craftcms/cms/src/web/Application.php(625): craft\web\Application->runAction('calendar/api/ic...', Array)
#11 vendor/craftcms/cms/src/web/Application.php(280): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#12 vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#13 web/index.php(14): yii\base\Application->run()
#14 {main}
```

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.2.1.1
- Calendar Version: 4.0.0-beta.8
- Calendar Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Comment by kjmartens** [comment]
Sorry for the delay @gglnx,

Is this error happening when using `{{ calendar.icsUrl }}` in a template? Or is it happening when attempting to visit that URL?  I'm curious what you have configured for your ICS Location and ICS Description fields for the calendar as well. Are you using a regular text/textarea field for these, or a special field type (the latter is not currently supported)? 🙂

**Comment by gglnx** [comment]
@kjmartens Sorry, I've forgot to answer. The exception occors when visiting the ICS URL. Location is a textarea field and the description is a redactor field.

**Comment by gglnx** [comment]
@kjmartens It's missing a check if rrule is null.

**Comment by kjmartens** [comment]
Sorry for the delay, @gglnx. I am able to duplicate this issue now. We'll get a fix for it shortly. 🙂

**Comment by kjmartens** [comment]
This should be resolved in Calendar 4.0.2 now 🙂
